### PR TITLE
Add bobcat jobs and charm-functional-jobs

### PR DESCRIPTION
**Pull request by coreycb**
Add bobcat jobs and charm-functional-jobs for bobcat.

Also stop running zed jobs on master branches.

### CODE REVIEW

File: zuul.d/project-templates.yaml [hunk 1]
```diff
@@ -76,6 +76,23 @@
       The default set of functional test jobs for the OpenStack Charms
     check:
       jobs:
+        - mantic-bobcat:
+            voting: false
+            branches:
+              - master
+              - stable/2023.2
+              - stable/1.8
+              - stable/23.03
```
Comment (freyes) [inline zuul.d/project-templates.yaml:85]: ovn would be stable/23.09 , upstream has a 3 months cadence, the charms align with the March and September releases.

**Review by freyes — CHANGES_REQUESTED** [review]

File: zuul.d/project-templates.yaml (hunk 1 above)
Comment (coreycb) [reply zuul.d/project-templates.yaml]: Thanks, I've added a new commit for this

File: zuul.d/project-templates.yaml [hunk 2]
```diff
@@ -76,6 +76,23 @@
...
+              - stable/quincy.2
```
[the remark this answers is not in the document]
Comment (coreycb) [reply zuul.d/project-templates.yaml]: New commit incoming with this change.

**Review by freyes — APPROVED** [review]
LGTM

**Pull request merged**